packages/compass-components/src/components/document-list/element-editors.tsx

```diff
@@ -333,33 +347,39 @@ export const TypeEditor: React.FunctionComponent<{
   return (
     <>
       {editing && (
-        // This rule is deprecated
-        // https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-onchange.md#deprecated-no-onchange
-        // eslint-disable-next-line jsx-a11y/no-onchange
-        <select
-          value={type}
-          data-testid="hadron-document-type-editor"
-          // See ./element.tsx
-          // eslint-disable-next-line jsx-a11y/no-autofocus
-          autoFocus={autoFocus}
-          onChange={(evt) => {
-            onChange(evt.currentTarget.value as HadronElementType['type']);
-          }}
-          className={cx(
-            editorReset,
-            editorOutline,
-            typeEditor,
-            visuallyActive && typeEditorActive
-          )}
-        >
-          {TYPES.map((type) => {
-            return (
-              <option key={type} value={type}>
-                {type}
-              </option>
-            );
-          })}
-        </select>
+        <div className={typeEditorContainer}>
+          {/* This rule is deprecated https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-onchange.md#deprecated-no-onchange */}
+          {/* eslint-disable-next-line jsx-a11y/no-onchange */}
+          <select
+            value={type}
+            data-testid="hadron-document-type-editor"
+            // See ./element.tsx
+            // eslint-disable-next-line jsx-a11y/no-autofocus
+            autoFocus={autoFocus}
+            onChange={(evt) => {
+              onChange(evt.currentTarget.value as HadronElementType['type']);
+            }}
+            className={cx(
+              editorReset,
+              editorOutline,
+              typeEditor,
+              visuallyActive && typeEditorActive
+            )}
```

Comment:
Unrelated, something that's already in main, if a user clicks on the select type dropdown on the document view on a field that is an array or an object it will expand or collapse the field. If we stop the click propagating here we can make that ux a bit nicer.
```suggestion
            )}
            onClick={e => {e.stopPropagation();}}
```